cfripper/rules/stack_name_matches_regex.py

```diff
@@ -0,0 +1,46 @@
+from typing import Dict, Optional
+
+from pycfmodel.model.cf_model import CFModel
+
+from cfripper.config.regex import REGEX_ALPHANUMERICAL_OR_HYPHEN
+from cfripper.model.enums import RuleGranularity, RuleMode, RuleRisk
+from cfripper.model.result import Result
+from cfripper.rules.base_rules import Rule
+
+
+class StackNameMatchesRegexRule(Rule):
+    """
+    Checks that a given stack follows the naming convention given by a regex. For this to work,
+    the stack name must be given either in the config or in the extras using the key
+    "stack_name".
+    """
+
+    RULE_MODE = RuleMode.DEBUG  # for demonstration purposes
+    RISK_VALUE = RuleRisk.LOW
+    GRANULARITY = RuleGranularity.STACK
+    REASON = (
+        "The stack name {} does not follow the naming convention (only alphanumerical characters and hyphens allowed)."
+    )
+    REGEX = REGEX_ALPHANUMERICAL_OR_HYPHEN
```

Comment:
I would suggest that if you are assigning regex in this way to show it can be any value you should similarly have a value for the reason. 
`REASON = (f"The Stack name {} does not follow the naming convention {REGEX_REASON}` where you assign the regex reason separately.